client/components/add-new-site/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { useBreakpoint } from '@automattic/viewport-react';
+import { useRef, useState, useCallback, useMemo } from 'react';
+import isA8CForAgencies from 'calypso/lib/a8c-for-agencies/is-a8c-for-agencies';
+import AddNewSiteButton from './button';
+import { AddNewSiteContext } from './context';
+import AddNewSiteMenuItems from './menu-items';
+import AddNewSiteModals from './modals';
+import AddNewSitePopover from './popover';
+
+import './style.scss';
+
+const AddNewSite = () => {
+	const isNarrowView = useBreakpoint( '<660px' );
+
+	const [ isMenuVisible, setMenuVisible ] = useState( false );
+	const [ visibleModalType, setVisibleModalType ] = useState( '' );
+
+	const popoverMenuContext = useRef( null );
+
+	const toggleMenu = useCallback( () => {
+		setMenuVisible( ( isVisible ) => ! isVisible );
+	}, [] );
+
+	// Render the popover content based on the environment
+	const renderPopoverContent = useMemo( () => {
+		switch ( true ) {
+			case isA8CForAgencies():
+				return <AddNewSiteMenuItems.A4A setMenuVisible={ setMenuVisible } />;
```

Comment:
We should probably figure out how to asynchronously load this part, so we don't force non-A4A consumers to import code that is not relevant to their use case. 🤔  

Honestly, I'm not sure if there is any benefit in putting everything in one central place rather than letting each consumer (WPOM, Jetpack or A4A) implement their own 'Add new site' component.

Reply:
I have made the changes to load the modal asynchronously. Loading the menu items on load is a bit complicated in this case as the component gets rendered when the button is clicked. I think it's not a large piece of code, so it ok to load this synchronously as we need it to be available as soon as the button is clicked without compromising on the user experience.

On the other hand, I have implemented async loading for the modal here. So, that won't get loaded if unnecessary.